src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -183,3 +172,9 @@ def print_revert_report(self, *, delete_managed: bool) -> bool | None:
             print("Migrated Manged Tables (targets) will be left intact.")
             print("To revert and delete Migrated Tables, add --delete_managed true flag to the command.")
         return True
+
+    def _get_mapping_rules_dict(self) -> dict[str, Rule]:
+        mapping_rules: dict[str, Rule] = {}
+        for rule in self._tm.load():
+            mapping_rules[rule.as_hms_table_key] = rule
```

Comment:
Add the following steps:

1. get `upgraded_to` property from `rule.as_hms_table_key()` table in HMS
2. get `skipped` property from `rule.as_hms_table_key()` table in HMS
3. get `upgraded_from` property from `rule.as_uc_table_key()` table in UC

Create new dataclass

```
@dataclass
class TableToMigrate:
  rule: Rule
  src: Table
  dst: TableInfo | None = None
  upgraded_to_on_src: bool = False
  
  def upgraded_from_on_dst(self): if not self.dst: return None else return self.dst.properties['upgraded_from']
  def is_migrated(self): return self.dst is not None
  def target_table_name(self): return f'`{self.rule.catalog}`.`{self.rule.dst_schema}`.`{self.rule.dst_table}`'
```

Reply:
https://github.com/databrickslabs/ucx/issues/754